app/questionnaire/questionnaire_schema.py

```diff
@@ -285,14 +282,45 @@ def _populate_answer_dependencies(self) -> None:
                                 option["detail_answer"], block_id=block["id"]
                             )
 
-    def _update_answer_dependencies_for_calculated_summary(
-        self, calculated_summary_answer_ids: Iterable[str], block_id: str
+    def _update_answer_dependencies_for_summary(self, block: ImmutableDict) -> None:
+        if block["type"] == "CalculatedSummary":
+            self._update_answer_dependencies_for_calculated_summary_dependency(
+                calculated_summary_block=block, dependent_block=block
+            )
+        else:  # must be GrandCalculatedSummary
+            self._update_answer_dependencies_for_grand_calculated_summary(block)
+
+    def _update_answer_dependencies_for_calculated_summary_dependency(
+        self, *, calculated_summary_block: ImmutableDict, dependent_block: ImmutableDict
     ) -> None:
+        """
+        Given a calculated summary block and a block which depends on that calculated summary (may just be the calculated summary itself)
+        update the answer to be a dependency for the dependent block
```

Comment:
If I'm understanding the function correctly this should be:

"update all calculated summary answers to be dependencies of the dependent block"

Reply:
You are indeed understanding and thanks, thats a better more concise comment - updated